javalanglib/src/main/scala/java/lang/Math.scala

```diff
@@ -219,5 +219,110 @@ object Math {
     }
   }
 
-  // TODO The methods not available in the JavaScript Math object
+  def addExact(a: scala.Int, b: scala.Int): scala.Int = {
+    val res = a + b
+    val resSgnBit = res < 0
+    if (resSgnBit == (a < 0) || resSgnBit == (b < 0)) res
+    else throw new ArithmeticException("Integer overflow")
+  }
+
+  def addExact(a: scala.Long, b: scala.Long): scala.Long = {
+    val res = a + b
+    val resSgnBit = res < 0
+    if (resSgnBit == (a < 0) || resSgnBit == (b < 0)) res
+    else throw new ArithmeticException("Long overflow")
+  }
+
+  def subtractExact(a: scala.Int, b: scala.Int): scala.Int = {
+    val res = a - b
+    val resSgnBit = res < 0
+    if (resSgnBit == (a < 0) || resSgnBit == (b > 0)) res
+    else throw new ArithmeticException("Integer overflow")
+  }
+
+  def subtractExact(a: scala.Long, b: scala.Long): scala.Long = {
+    val res = a - b
+    val resSgnBit = res < 0
+    if (resSgnBit == (a < 0) || resSgnBit == (b > 0)) res
+    else throw new ArithmeticException("Long overflow")
+  }
+
+  def multiplyExact(a: scala.Int, b: scala.Int): scala.Int = {
+    val overflow = {
+      if (b > 0)
+        a > Integer.MAX_VALUE / b || a < Integer.MIN_VALUE / b
+      else if (b < -1)
+        a > Integer.MIN_VALUE / b || a < Integer.MAX_VALUE / b
+      else if (b == -1)
+        a == Integer.MIN_VALUE
+      else
+        false
+    }
+    if (!overflow) a * b
+    else throw new ArithmeticException("Integer overflow")
+  }
+
+  def multiplyExact(a: scala.Long, b: scala.Long): scala.Long = {
+    val overflow = {
+      if (b > 0)
+        a > Long.MAX_VALUE / b || a < Long.MIN_VALUE / b
+      else if (b < -1)
+        a > Long.MIN_VALUE / b || a < Long.MAX_VALUE / b
+      else if (b == -1)
+        a == Long.MIN_VALUE
+      else
+        false
+    }
+    //if (!overflow) a * b // Disabled because of #1918
```

Comment:
For this, I think we should wait for #1923 to be merged, then rebase this PR and adapt this.